test/distributed/_composable/test_compose.py

```diff
@@ -145,6 +147,33 @@ def test_checkpoint_fsdp_submodules_use_reentrant(self):
     def test_checkpoint_fsdp_submodules_non_reentrant(self):
         self._test_checkpoint_fsdp_submodules(False)
 
+    @skip_if_lt_x_gpu(2)
+    def test_fully_shard_replicate_correct_replicate_params(self):
+        model = CompositeParamModel(device=torch.device("cuda"))
+        # Shard Linears within UnitModule
+        fully_shard(model.u1, policy=ModuleWrapPolicy({nn.Linear}))
+        fully_shard(model.u2, policy=ModuleWrapPolicy({nn.Linear}))
+        # replicate the rest
+        replicate(model)
+        # Run fwd + bwd to initialize DDP
+        inp = torch.randn(2, 100, device="cuda")
+        model(inp).sum().backward()
+        # Ensure replicate param names are as expected, i.e.
+        # immediate parameters of model and parameters of model's non-UnitModule
+        # submodules are replicated
+        param_names = replicate.state(model)._replicate_param_names
+        replicated_modules = [
+            (name, mod)
+            for (name, mod) in model.named_children()
+            if mod not in [model.u1, model.u2]
+        ]
+        replicated_param_names = [
+            f"{module_name}.{n}"
```

Comment:
So these are _not_ all prefixed from the same root module?

Reply:
For something like
```
class MyModel:
    def __init__(self):
        self.lin1 = ...
        self.lin2 = ....
        self.p = nn.Parameter(...)
```

and if we replicate the whole thing, the names we expect are like `[lin1.weight, lin2.weight, p]`. They are all prefixed from the same root, but the root is `MyModel` and the root prefix is just `""`.